### PR TITLE
26 initial inspected points not counted towards reward

### LINKED ISSUE #26: initial inspected points not counted towards reward
on env reset self.prev_weight_inspected is set to 0.0 but on each step, it is set to self.chief.inspection_points.get_total_weight_inspected() before taking any action. So any points seen on initialization will not be included in the points reward. So in the end, the points reward will be total_weight - initial_weight, often significantly less than the success threshold since it sees a lot of points at first.

### PR DESCRIPTION
**Pull request by keatincf**
The fix for the issue was relatively simple, but there were a lot of mypy, flake8, and pylint warnings/errors that needed to be resolved before I could commit.

### CODE REVIEW

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""Unit tests for rewards for the Inspection Gymnasium environment"""
+
+import pytest
+from safe_autonomy_simulation.sims.inspection.target import Target
+
+from safe_autonomy_sims.gym.inspection.reward import observed_points_reward, weighted_observed_points_reward
+
+
+def testObservedPointsRewardEnforcesInspectedPointsCannotBeUninspected():
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: python functions should be snake_case

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    target = Target("test_target", 2, 1)
+
+    try:
+        observed_points_reward(target, 1)
+        pytest.fail("no assertion error raised when observed point is no longer observed")
+    except AssertionError:
+        pass
+
+
+def testWeightedObservedPointsRewardEnforcesInspectedPointsCannotBeUninspected():
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: python functions should be snake_case

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_sixdof_inspection_v0.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+"""Unit tests for the weighted inspection environment"""
+
+from test.unit_tests.safe_autonomy_sims.mock import MockInspectionPointSet, StaticDynamics
+
+from safe_autonomy_sims.gym.inspection.sixdof_inspection_v0 import WeightedSixDofInspectionEnv
+
+
+def testWeighted6DOFInspectionEnvTerminatesOnInspectionThresholdMet():
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_sixdof_inspection_v0.py]: python functions should be snake_case

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_weighted_inspection_v0.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+"""Unit tests for the weighted inspection environment"""
+
+from test.unit_tests.safe_autonomy_sims.mock import MockInspectionPointSet, StaticDynamics
+
+from safe_autonomy_sims.gym.inspection.weighted_inspection_v0 import WeightedInspectionEnv
+
+
+def testWeightedInspectionEnvTerminatesOnInspectionThresholdMet():
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_weighted_inspection_v0.py]: python functions should be snake_case

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+"""Unit tests for the weighted inspection environment"""
+
+from test.unit_tests.safe_autonomy_sims.mock import MockInspectionPointSet, StaticDynamics
+
+from safe_autonomy_sims.pettingzoo.inspection.sixdof_multi_inspection_v0 import WeightedSixDofMultiInspectionEnv
+
+
+def testWeighted6DOFMultiInspectionEnvTerminatesOnInspectionThresholdMet():
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py]: python functions should be snake_case

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+"""Unit tests for the weighted inspection environment"""
+
+from test.unit_tests.safe_autonomy_sims.mock import MockInspectionPointSet, StaticDynamics
+
+from safe_autonomy_sims.pettingzoo.inspection.weighted_multi_inspection_v0 import WeightedMultiInspectionEnv
+
+
+def testWeightedMultiInspectionEnvTerminatesOnInspectionThresholdMet():
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py]: python functions should be snake_case

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+def testObservedPointsRewardEnforcesInspectedPointsCannotBeUninspected():
+    """Given a target being inspected, when determining the reward for inspected points and the target
+    has fewer points inspected than previously, then an assertion error is raised"""
+    target = Target("test_target", 2, 1)
+
+    try:
+        observed_points_reward(target, 1)
+        pytest.fail("no assertion error raised when observed point is no longer observed")
+    except AssertionError:
+        pass
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: This looks fine but pytest has a more succinct syntax for these types of tests that allows for more flexibility:

```python
with pytest.raises(MyError):
    my_failing_function(...)
```

See [the pytest documentation](https://docs.pytest.org/en/stable/how-to/assert.html#assertions-about-expected-exceptions) for more info.

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+def testWeightedObservedPointsRewardEnforcesInspectedPointsCannotBeUninspected():
+    """Given a target being inspected, when determining the reward for weighted inspected points and the target
+    has fewer points inspected than previously, then an assertion error is raised"""
+    target = Target("test_target", 2, 1)
+
+    try:
+        weighted_observed_points_reward(target, 0.5)
+        pytest.fail("no assertion error raised when observed point is no longer observed")
+    except AssertionError:
+        pass
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: This looks fine but pytest has a more succinct syntax for these types of tests that allows for more flexibility:

```python
with pytest.raises(MyError):
    my_failing_function(...)
```

See [the pytest documentation](https://docs.pytest.org/en/stable/how-to/assert.html#assertions-about-expected-exceptions) for more info.

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_sixdof_inspection_v0.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    env.reset()
+
+    # This isn't great because we're modifying the internal via a non-public interface, but Python allows it
+    # To do this via a public interface, we'd either have to expose things we don't actually want to expose
+    # just for testing or rearchitect to allow more modularity, but increases complexity.  Neither option
+    # is particularly great right now, so we can live with swapping out internals.  However, this will make
+    # this test brittle if the internal workings of the environment changes, such as lazy initialization
+    # being done by some other func that is called after reset.
+    env.chief._inspection_points = MockInspectionPointSet(parent=env.chief)  # pylint: disable=W0212
+    env.deputy._dynamics = StaticDynamics()  # pylint: disable=W0212
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_sixdof_inspection_v0.py]: These may be worth patching. The `pytest-mock` package allows for using mocks with pytest fixtures. A simpler approach might be the native `unittest.mock` package which should work just fine with pytest.

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_weighted_inspection_v0.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    env.reset()
+
+    # This isn't great because we're modifying the internal via a non-public interface, but Python allows it
+    # To do this via a public interface, we'd either have to expose things we don't actually want to expose
+    # just for testing or rearchitect to allow more modularity, but increases complexity.  Neither option
+    # is particularly great right now, so we can live with swapping out internals.  However, this will make
+    # this test brittle if the internal workings of the environment changes, such as lazy initialization
+    # being done by some other func that is called after reset.
+    env.chief._inspection_points = MockInspectionPointSet(parent=env.chief)  # pylint: disable=W0212
+    env.deputy._dynamics = StaticDynamics()  # pylint: disable=W0212
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/gym/inspection/test_weighted_inspection_v0.py]: These may be worth patching. The `pytest-mock` package allows for using mocks with pytest fixtures. A simpler approach might be the native `unittest.mock` package which should work just fine with pytest.

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    env = WeightedSixDofMultiInspectionEnv()
+    env.reset()
+
+    # This isn't great because we're modifying the internal via a non-public interface, but Python allows it
+    # To do this via a public interface, we'd either have to expose things we don't actually want to expose
+    # just for testing or rearchitect to allow more modularity, but increases complexity.  Neither option
+    # is particularly great right now, so we can live with swapping out internals.  However, this will make
+    # this test brittle if the internal workings of the environment changes, such as lazy initialization
+    # being done by some other func that is called after reset.
+    env.chief._inspection_points = MockInspectionPointSet(parent=env.chief)  # pylint: disable=W0212
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py]: These may be worth patching. The `pytest-mock` package allows for using mocks with pytest fixtures. A simpler approach might be the native `unittest.mock` package which should work just fine with pytest.

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    env = WeightedMultiInspectionEnv()
+    env.reset()
+
+    # This isn't great because we're modifying the internal via a non-public interface, but Python allows it
+    # To do this via a public interface, we'd either have to expose things we don't actually want to expose
+    # just for testing or rearchitect to allow more modularity, but increases complexity.  Neither option
+    # is particularly great right now, so we can live with swapping out internals.  However, this will make
+    # this test brittle if the internal workings of the environment changes, such as lazy initialization
+    # being done by some other func that is called after reset.
+    env.chief._inspection_points = MockInspectionPointSet(parent=env.chief)  # pylint: disable=W0212
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py]: These may be worth patching. The `pytest-mock` package allows for using mocks with pytest fixtures. A simpler approach might be the native `unittest.mock` package which should work just fine with pytest.

**Review by jamie-cunningham — CHANGES_REQUESTED** [review]
Good job on this. I left a few requested changes and some feedback. I really appreciate the handling of the mypy/formatting stuff. Some of it seems configured strangely so it may be worth the lift to configure those checks properly soon.

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py (hunk 1 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: Updated name to be snake_case

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py (hunk 2 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: Updated name to be snake_case

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py (hunk 5 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py]: Updated name to be snake_case

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py (hunk 6 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py]: Updated name to be snake_case

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py (hunk 12 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_weighted_multi_inspection_v0.py]: Refactored to patch w/ mock class

File: test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py (hunk 11 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/pettingzoo/inspection/test_sixdof_multi_inspection_v0.py]: Refactored to patch w/ mock class

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_weighted_inspection_v0.py (hunk 10 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/gym/inspection/test_weighted_inspection_v0.py]: Refactored to patch w/ mock class

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_sixdof_inspection_v0.py (hunk 9 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/gym/inspection/test_sixdof_inspection_v0.py]: Refactored to patch w/ mock class

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py (hunk 8 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: Changed to the more succinct syntax

File: test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py (hunk 7 above)
Comment (keatincf) [reply test/unit_tests/safe_autonomy_sims/gym/inspection/test_reward.py]: Changed to the more succinct syntax

**Review by jamie-cunningham — APPROVED** [review]

**Pull request merged**